port/osx/omrintrospect.c

```diff
@@ -405,19 +506,25 @@ omrintrospect_threads_nextDo(J9ThreadWalkState *state)
 	freeThread(state, state->current_thread);
 
 	data->threadIndex += 1;
-	if (data->filterThread == data->threadList[data->threadIndex]) {
-		data->threadIndex += 1;
-	}
-
 	if (data->threadIndex == data->threadCount) {
 		/* Finished processing threads. */
-		return NULL;
+		goto cleanup;
 	}
 
 	result = setupNativeThread(state, NULL);
 	if (0 != result) {
+		if (NULL == state->current_thread) {
+			/* Allocation failure, can only return NULL which terminates iteration. */
+			goto cleanup;
+		} else if (TIMEOUT == result) {
+			/* Timeout, don't walk rest of threads. Wait for thread in upcallHandler to complete before
+			 * freeing thread. */
```

Comment:
should be on a new line.
```suggestion
			 * freeing thread.
			 */
```